src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/LogsTests.cs

```diff
@@ -0,0 +1,638 @@
+
+            // Set up a channel and process the log events here rather than having each test have to deserialize
+            // the set of log events. Pass the channel reader to the callback to allow each test to verify the
+            // set of deserialized log events.
+            Channel<LogEntry> channel = Channel.CreateUnbounded<LogEntry>(new UnboundedChannelOptions()
+            {
+                SingleReader = true,
+                SingleWriter = true,
+                AllowSynchronousContinuations = false
+            });
+
+            Task callbackTask = callback(channel.Reader);
+
+            using StreamReader reader = new StreamReader(holder.Stream);
+
+            JsonSerializerOptions options = new();
+            options.Converters.Add(new JsonStringEnumConverter());
+
+            string line;
+            while (null != (line = await reader.ReadLineAsync()))
+            {
+                _outputHelper.WriteLine("Log entry: {0}", line);
+                try
+                {
+                    await channel.Writer.WriteAsync(JsonSerializer.Deserialize<LogEntry>(line, options));
+                }
+                catch (JsonException ex)
+                {
+                    _outputHelper.WriteLine("Exception while deserializing log entry: {0}", ex);
+                }
+            }
+            channel.Writer.Complete();
+
+            await callbackTask;
+        }
+
+        /// <summary>
+        /// Validates each aspect of a <see cref="LogEntry"/> compared to the expected values
+        /// on a reference <see cref="LogEntry"/> instance.
+        /// </summary>
+        /// <remarks>
+        /// The Exception property of the <paramref name="expected"/> argument is compared
+        /// to the first line of the Exception property of the <paramref name="actual"/> argument,
+        /// which contains the exception name and message.
+        /// </remarks>
+        private static void ValidateEntry(LogEntry expected, LogEntry actual)
+        {
+            Assert.Equal(expected.Category, actual.Category);
+
+            Assert.Equal(expected.EventId, actual.EventId);
+            
+            Assert.Equal(expected.EventName, actual.EventName);
+
+            if (null == expected.Exception)
+            {
+                Assert.Null(actual.Exception);
+            }
+            else
+            {
+                Assert.NotNull(actual.Exception);
+                // Only compare the first line, which contains the exception type and message.
+                // The other lines contain callstack information, which will vary between machines.
+                string firstLine = actual.Exception.Split(Environment.NewLine)[0];
+                Assert.Equal(expected.Exception, firstLine);
+            }
+
+            Assert.Equal(expected.LogLevel, actual.LogLevel);
+
+            Assert.Equal(expected.Message, actual.Message);
+
+            // TODO: Work on scope verification
```

Comment:
I'd like to add scope information to the log events of the target application, however I'll do that in a subsequent change.